scrapy/utils/misc.py

```diff
@@ -137,17 +137,24 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
     ``*args`` and ``**kwargs`` are forwarded to the constructors.
 
     Raises ``ValueError`` if both ``settings`` and ``crawler`` are ``None``.
+
+    Raises ``TypeError`` if the resulting instance is ``None`` (e.g. if an
+    extension has not been implemented correctly).
     """
     if settings is None:
         if crawler is None:
             raise ValueError("Specify at least one of settings and crawler.")
         settings = crawler.settings
+    instance = None
```

Comment:
This is not necessary, the below `if` statement covers all scenarios

Reply:
Oh yeah, thanks, will change. I'd forgotten python isn't block scoped.